src/function/scalar/list/list_select.cpp

```diff
@@ -0,0 +1,103 @@
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/planner/expression_binder.hpp"
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/planner/expression/bound_parameter_expression.hpp"
+
+namespace duckdb {
+
+static void ListSelectFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	Vector &list = args.data[0];
+	Vector &selection_list = args.data[1];
+	idx_t count = args.size();
+
+	ListVector::Reserve(result, ListVector::GetListSize(selection_list));
+
+	bool all_constant = true;
+	for (idx_t i = 0; i < args.ColumnCount(); i++) {
+		auto curr = args.data[i];
+		// check if all arguments are constants, if so the result is also a constant vector
+		if (curr.GetVectorType() != VectorType::CONSTANT_VECTOR) {
+			all_constant = false;
+		}
+	}
+
+	list_entry_t *result_data;
+	result_data = FlatVector::GetData<list_entry_t>(result);
+	auto &result_entry = ListVector::GetEntry(result);
+
+	UnifiedVectorFormat sel_list;
+	selection_list.ToUnifiedFormat(count, sel_list);
+	auto selection_lists = UnifiedVectorFormat::GetData<list_entry_t>(sel_list);
+	auto &sel_entry = ListVector::GetEntry(selection_list);
+
+	UnifiedVectorFormat inp_list;
```

Comment:
Nit: `input` instead of `inp`? The name is not too long, so there is no need to abbreviate it. The same holds for later variable names.